src/main/java/sirius/biz/jupiter/JupiterSync.java

```diff
@@ -164,4 +215,249 @@ private String asYaml(Map<String, Object> config) {
+            String url = store.objectUrl(uplinkBucketName, object.getKey());
+            processContext.log(ProcessLog.info()
+                                         .withFormattedMessage("Fetching %s for %s as it is new or updated...",
+                                                               effectiveFileName,
+                                                               connection.getName()));
+            connection.repository().fetchUrl(effectiveFileName, url, false);
+        } else {
+            processContext.debug(ProcessLog.info()
+                                           .withFormattedMessage("Skipping %s for %s as it is unchanged...",
+                                                                 effectiveFileName,
+                                                                 connection.getName()));
+        }
+
+        filesToDelete.remove(effectiveFileName);
+    }
+
+    private void syncLocalRepository(ProcessContext processContext,
+                                     JupiterConnector connection,
+                                     List<RepositoryFile> repositoryFiles,
+                                     Set<String> filesToDelete) {
+        if (blobStorage == null || tenants == null || Strings.isEmpty(localRepoSpaceName)) {
+            processContext.debug(ProcessLog.info()
+                                           .withFormattedMessage(
+                                                   "Skipping local repository for %s as no storage space is configured.",
+                                                   connection.getName()));
+            return;
+        }
+
+        processContext.debug(ProcessLog.info()
+                                       .withFormattedMessage("Checking local repository in storage space %s for %s...",
+                                                             localRepoSpaceName,
+                                                             connection.getName()));
+
+        try {
+            Directory root = blobStorage.getSpace(localRepoSpaceName)
+                                        .getRoot(tenants.getTenantUserManager().getSystemTenantId());
+            visitLocalDirectory(processContext, "", root, connection, repositoryFiles, filesToDelete);
+        } catch (Exception e) {
+            processContext.handle(Exceptions.handle()
+                                            .error(e)
+                                            .withSystemErrorMessage(
+                                                    "Failed to check the local repository %s for %s: %s (%s)",
+                                                    localRepoSpaceName,
+                                                    connection.getName())
+                                            .handle());
+        }
+    }
+
+    private void visitLocalDirectory(ProcessContext processContext,
+                                     String prefix,
```

Comment:
make this nullable? So we dont have to pass "" above

Reply:
nope null + "/" != "" + "/"

Reply:
Yes my intention was to suggest handling a passed null inside the method with a simple if or ternary